flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/RollingSink.java

```diff
@@ -533,26 +533,34 @@ private Method reflectTruncate(FileSystem fs) {
 			}
 
 			// verify that truncate actually works
-			Path testPath = new Path(UUID.randomUUID().toString());
-			try (FSDataOutputStream outputStream = fs.create(testPath)) {
-				outputStream.writeUTF("hello");
-			} catch (IOException e) {
-				LOG.error("Could not create file for checking if truncate works.", e);
-				throw new RuntimeException("Could not create file for checking if truncate works.", e);
-			}
-
+			Path testPath = new Path(basePath, UUID.randomUUID().toString());
 			try {
-				m.invoke(fs, testPath, 2);
-			} catch (IllegalAccessException | InvocationTargetException e) {
-				LOG.debug("Truncate is not supported.", e);
-				m = null;
-			}
+				try (FSDataOutputStream outputStream = fs.create(testPath)) {
+					outputStream.writeUTF("hello");
+				} catch (IOException e) {
+					LOG.error("Could not create file for checking if truncate works.", e);
+					throw new RuntimeException(
+							"Could not create file for checking if truncate works. " +
+									"You can disable support for truncate() completely via " +
+									"BucketingSink.setUseTruncate(false).",
+							e);
+				}
 
-			try {
-				fs.delete(testPath, false);
-			} catch (IOException e) {
-				LOG.error("Could not delete truncate test file.", e);
-				throw new RuntimeException("Could not delete truncate test file.", e);
+				try {
+					m.invoke(fs, testPath, 2);
+				} catch (IllegalAccessException | InvocationTargetException e) {
+					LOG.debug("Truncate is not supported.", e);
+					m = null;
+				}
+			} finally {
+				try {
+					fs.delete(testPath, false);
+				} catch (IOException e) {
+					LOG.error("Could not delete truncate test file.", e);
+					throw new RuntimeException("Could not delete truncate test file. " +
```

Comment:
I am not sure if we want to fail the job if we cannot `delete` the `test` file.
I would say it is enough to just log the fact and move on. If there is a bigger problem and this was not just a hick-up, the job will fail later when something that matters fails (e.g. we cannot create a tmp file).

What do you think @aljoscha ?

Reply:
I agree, but I want to keep this change minimal. The decision a while back was to throw this exception and have users manually disable truncate if they want to get rid of the exception.